libs/server/src/lib/telemetry/user.ts

```diff
@@ -0,0 +1,98 @@
+import { PersistentStorage } from '@angular-console/storage';
+import { authUtils } from '@nrwl/angular-console-enterprise-electron';
+
+import * as uuid from 'uuid/v4';
+
+export type UserState = 'untracked' | 'anonymous' | 'connected';
+
+interface Settings {
+  isConnectUser: boolean;
+  canCollectData: boolean;
+}
+
+export type CheckConnection = () => boolean;
+export type OnChange = (user: User) => void;
+
+function isConnected(): boolean {
+  return !!authUtils.getIdTokenFromStore();
+}
+
+export class User {
+  static fromStorage(store: PersistentStorage): User {
+    authUtils.setStore(store);
+    const settings: Settings | null = store.get('settings');
+    let id: string | null = store.get('uuid');
+    let state: UserState = 'anonymous';
+
+    if (!id) {
+      id = uuid();
+      store.set('uuid', id);
+    }
+
+    const connected = authUtils.getIdTokenFromStore();
+    if (connected) {
+      state = 'connected';
+    }
+
+    if (settings && !settings.canCollectData) {
+      state = 'untracked';
+    }
+
+    const user = new User(id, state, isConnected);
+    return user;
+  }
+
+  onChange: OnChange = () => {};
```

Comment:
Why leave this method in?

Reply:
it is used here https://github.com/nrwl/angular-console/pull/801/files#diff-1911b3879856aa4f2cc5d325f594e7beR39

Reply:
It is used in the google analytics sink to respond to logged in state changing.

Reply:
If you are implementing an interface or abstract class could you annotate this class with that information?

Reply:
Changed the way that user works. The idea of `User` is that it handles the state machine around user state relating to telemetry. I had the google sink reading from it directly, I think the better idea is that User is something telemetry uses, and turns into another type of record. That removes a collaboration, and it puts that in the central mechanism of telemetry writing records, think its a better design :)